examples/image_warping_example.py

```diff
@@ -0,0 +1,83 @@
+"""
+Mouse event callbacks
+=====================
+
+Warp an image based on points moved by user using mouse event callbacks in
+napari.
+
+This example is fully explained in the following tutorial:
+https://napari.org/napari-scipy2025-workshop/notebooks/image_warping.html
+
+.. tags:: interactivity
+"""
+
+from functools import partial
+
+import numpy as np
+import skimage as ski
+
+import napari
+
+# Set up base image (to be warped) and points layers
+image = ski.data.checkerboard()
+src = np.array(
+    [
+        [ 66, 66],
+        [133,  66],
+        [ 66, 133],
+        [133, 133]
+    ]
+)
+
+viewer = napari.Viewer()
+checkerboard_image_layer = viewer.add_image(image, name='checkerboard')
+source_points_layer = viewer.add_points(src, name='source_points', symbol='+', face_color='red', size=5)
+moving_points_layer = viewer.add_points(src.copy(), name='moving_points')
+
+
+# Define a function to estimate the warping required to transform the destination points
+# into the source points, and then apply the warping to the original image data,
+# replacing the image layer data in-place.
+def warp(im_layer: 'napari.layers.Image', src: 'np.ndarray', dst: 'np.ndarray') -> None:
+    # Warp image using thin-plate spline transformation from skimage.
+    tps = ski.transform.ThinPlateSplineTransform()
+    tps.estimate(dst, src)
+    warped = ski.transform.warp(image, tps)
```

Comment:
I assume this is from upstream 😬 but this `image` should properly be `im_layer.data` right? the data of the layer from the argument of the function?
```suggestion
    warped = ski.transform.warp(im_layer.data, tps)
```

Reply:
cc @DragaDoncila

Reply:
Oops yep, good catch Peter. The fix is actually a little trickier, because we need to make sure we're always warping on the original image data. So, I think the correct fix is to update the `warp` function to take the original image data as a separate argument, and then bind that below with the partial. So, `warp` would become:

```python
def warp(
    im_layer: 'napari.layers.Image',
    original_im_data: 'np.ndarray',
    src: 'np.ndarray',
    dst: 'np.ndarray'
) -> None:
    tps = ski.transform.ThinPlateSplineTransform()
    tps.estimate(dst, src)
    warped = ski.transform.warp(original_im_data, tps)
    im_layer.data = (warped * 255).astype(original_im_data.dtype)
```
Then the `original_im_data` must also be bound in the partial:

```python
# note `image` is passed through as well
warp_checkerboard = partial(warp, checkerboard_image_layer, image, src)
```